From c0c34b2d7eca73075e64eb01edf9b5447688ac4b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Tue, 24 Apr 2018 13:31:50 +0300
Subject: [PATCH 419/767] crypto: caam/qi2 - fix endianness of FD length
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

FD length is copied from input FLE length.
But since the FLE length has already been swapped to device endianness,
it needs to be swapped back to CPU endianness when accessed.

Fixes: 856f06d8d456 ("crypto: caam/qi2 - add DPAA2-CAAM driver")
Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/caamalg_qi2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/caamalg_qi2.c b/drivers/crypto/caam/caamalg_qi2.c
index 2cb702e8cb66..1153bfcdfdbf 100644
--- a/drivers/crypto/caam/caamalg_qi2.c
+++ b/drivers/crypto/caam/caamalg_qi2.c
@@ -5880,7 +5880,7 @@ int dpaa2_caam_enqueue(struct device *dev, struct caam_request *req)
 	memset(&fd, 0, sizeof(fd));
 	dpaa2_fd_set_format(&fd, dpaa2_fd_list);
 	dpaa2_fd_set_addr(&fd, req->fd_flt_dma);
-	dpaa2_fd_set_len(&fd, req->fd_flt[1].len);
+	dpaa2_fd_set_len(&fd, dpaa2_fl_get_len(&req->fd_flt[1]));
 	dpaa2_fd_set_flc(&fd, req->flc_dma);
 
 	/*
-- 
2.17.0

